main.tf

```diff
@@ -15,9 +19,9 @@ resource "aws_sns_topic" "this" {
 }
 
 resource "aws_sns_topic_subscription" "this" {
-  for_each = var.subscribers
+  for_each = local.enabled ? var.subscribers : {}
```

Comment:
@nitrocode  did you test it with `enabled=false`?
usually in this case terraform complains that both sides of the `?` operator must have the same types (not only it must be a map, but all items must be the same with the same types)

Reply:
> var.subscribers is a map so both types are the same

yes, but in similar cases Terraform complains that not only the container types must be the same (maps), but they must have the the items and the same item types

Reply:
What would you propose to use here instead ?